src/lb_data.c

```diff
@@ -1241,10 +1494,29 @@ int lb_halo_post(const lb_t * lb, lb_halo_t * h) {
 
   TIMER_start(TIMER_LB_HALO_PACK);
 
-  #pragma omp parallel
-  {
-    for (int ireq = 0; ireq < h->map.nvel; ireq++) {
-      lb_halo_enqueue_send(lb, h, ireq);
+  int ndevice;
+  tdpGetDeviceCount(&ndevice);
+  if (ndevice > 0 && lb->haloscheme == LB_HALO_TARGET) {
```

Comment:
We still may need to revisit haw to drive a host halo swap if `ndevice > 0`, but this is less important at the moment,.